into/backends/tests/test_aws.py

```diff
@@ -0,0 +1,253 @@
+import pytest
+import os
+import itertools
+from into import into, resource, S3, discover, CSV, drop, append, SSH
+from into.backends.aws import get_s3_connection
+from into.utils import tmpfile, filetext
+import pandas as pd
+import pandas.util.testing as tm
+import datashape
+
+sa = pytest.importorskip('sqlalchemy')
+boto = pytest.importorskip('boto')
+pytest.importorskip('psycopg2')
+pytest.importorskip('redshift_sqlalchemy')
+
+from boto.exception import S3ResponseError
+
+
+tips_uri = 's3://nyqpug/tips.csv'
+
+df = pd.DataFrame({
+    'a': list('abc'),
+    'b': [1, 2, 3],
+    'c': [1.0, 2.0, 3.0]
+})[['a', 'b', 'c']]
+
+
+js = pd.io.json.loads(pd.io.json.dumps(df, orient='records'))
+
+
+@pytest.fixture
+def db():
+    key = os.environ.get('REDSHIFT_DB_URI', None)
+    if not key:
+        pytest.skip('Please define a non-empty environment variable called '
+                    'REDSHIFT_DB_URI to test redshift <- S3')
+    else:
+        return key
+
+
+@pytest.yield_fixture
+def temp_tb(db):
+    t = '%s::%s' % (db, next(_tmps))
+    try:
+        yield t
+    finally:
+        drop(resource(t))
+
+
+@pytest.yield_fixture
+def tmpcsv():
+    with tmpfile('.csv') as fn:
+        with open(fn, mode='w') as f:
+            df.to_csv(f, index=False)
+        yield fn
+
+
+@pytest.fixture(scope='module')
+def conn():
+    # requires that you have a config file or envars defined for credentials
+    # this code makes me hate exceptions
+    try:
+        conn = get_s3_connection()
+    except S3ResponseError:
+        pytest.skip('unable to connect to s3')
+    else:
```

Comment:
Oh, or will Python keep chugging away on this function even after the pytest.skip